geomstats/geometry/pullback_metric.py

```diff
@@ -111,6 +113,53 @@ def pickable_inner_product(i, j):
         metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
         return metric_mat[0] if base_point.ndim == 1 else metric_mat
 
+    def _compute_hessian_immersion(self):
+        """Compute the Hessian of the immersion.
+
+        Returns
+        -------
+        hessian_immersion : list of callable
+            List of embedding_dim hessians of the scalar
+            functions defining the components of the immersion.
+        """
+
+        def _immersion(base_point, a):
+            """Compute the component a of the immersion at a base point.
+
+            Parameters
+            ----------
+            base_point : array-like, shape=[..., dim]
+                Base point.
+            a : int
+                Index of the component of the immersion.
+            """
+            return self.immersion(base_point)[a]
+
+        hessians = []
+        for a in range(self.embedding_metric.dim):
+            hessian_a = gs.autodiff.hessian(partial(_immersion, a=a))
+            hessians.append(hessian_a)
+        return hessians
+
+    def hessian_immersion(self, base_point):
+        """Compute the Hessian of the immersion.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        hessian_immersion : array-like, shape=[..., embedding_dim, dim, dim]
+            Hessian at the base point
+        """
+        if self._hessian_immersion is None:
+            self._hessian_immersion = self._compute_hessian_immersion()
+
+        hessian_values = [hes(base_point) for hes in self._hessian_immersion]
+        return gs.stack(hessian_values, axis=0)
+
     def inner_product_derivative_matrix(self, base_point=None):
         r"""Compute the inner-product derivative matrix.
```

Comment:
It would be nice to improve the docstrings by adding mathematical formulae, as for the case of similar implementations in `Connection` and `RiemannianMetric`.